paddle/fluid/operators/optimizers/sgd_op.cc

```diff
@@ -35,25 +35,20 @@ class SGDOp : public framework::OperatorWithKernel {
       const framework::ExecutionContext &ctx) const override {
     auto data_type = OperatorWithKernel::IndicateVarDataType(ctx, "Param");
 
-#ifdef PADDLE_WITH_MKLDNN
-    using dnnl::memory;
-    if (this->CanMKLDNNBeUsed(ctx, data_type)) {
-      const auto *param_var = ctx.InputVar("Param");
-      const auto *grad_var = ctx.InputVar("Grad");
-
-      // supported cases
-      bool dense_param_sparse_grad = param_var->IsType<phi::DenseTensor>() &&
-                                     grad_var->IsType<phi::SelectedRows>();
-      bool dense_param_and_grad = param_var->IsType<phi::DenseTensor>() &&
-                                  grad_var->IsType<phi::DenseTensor>();
-
-      if (dense_param_sparse_grad || dense_param_and_grad)
-        return framework::OpKernelType(data_type,
-                                       ctx.GetPlace(),
-                                       phi::DataLayout::kMKLDNN,
-                                       framework::LibraryType::kMKLDNN);
+    // NOTE(jiahongyu): Below codes originally enclosed by PADDLE_WITH_MKLDNN
+    const auto *param_var = ctx.InputVar("Param");
+    const auto *grad_var = ctx.InputVar("Grad");
+
+    // supported cases
+    bool dense_param_sparse_grad = param_var->IsType<phi::DenseTensor>() &&
+                                   grad_var->IsType<phi::SelectedRows>();
+    bool dense_param_and_grad = param_var->IsType<phi::DenseTensor>() &&
+                                grad_var->IsType<phi::DenseTensor>();
+    if (!(dense_param_sparse_grad || dense_param_and_grad)) {
+      dnn_fallback_ = true;
```

Comment:
Maybe adding a `set_dnn_fallback` method is better? Changing the member of base class in method of child class is not a good impl

Reply:
Done, thanks~